File: apps/frontend/src/types/CompraCartao.ts
export interface CreateCompraCartaoRequest {
    userId: string;
    descricao: string;
    valor: number;
    dataCompra: string; 
    categoria: string;
    cartao: string;
    parcelas: number;
}

export interface UpdateCompraCartaoRequest {
    id: string;
    descricao: string;
    valor: number;
    dataCompra: string;
    categoria: string;
    cartao: string;
    parcelas: number;
    status: 'Pendente' | 'Pago';
}

export interface CompraCartaoDto {
    id: string;
    descricao: string;
    valor: number;
    dataCompra: string;
    categoria: string;
    cartao: string;
    parcelas: number;
    createdAt: string;
}

File: apps/backend/src/FinanceManager.Domain/Entities/CompraCartao.cs
using System;

namespace FinanceManager.Domain.Entities
{
    public class CompraCartao
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public string Descricao { get; set; } = default!;
        public decimal Valor { get; set; }
        public DateTime DataCompra { get; set; }
        public string Categoria { get; set; } = default!;
        public string Cartao { get; set; } = default!;      
        public int Parcelas { get; set; }                    
        public DateTime CreatedAt { get; set; }
    }
}
